elixir/apps/domain/lib/domain/flows.ex

```diff
@@ -0,0 +1,88 @@
+defmodule Domain.Flows do
+  alias Domain.Repo
+  alias Domain.{Auth, Clients, Gateways, Resources, Policies}
+  alias Domain.Flows.{Authorizer, Flow}
+  require Ecto.Query
+
+  def authorize_flow(
+        %Clients.Client{
+          id: client_id,
+          account_id: account_id,
+          actor_id: actor_id,
+          identity_id: identity_id
+        },
+        %Gateways.Gateway{
+          id: gateway_id,
+          last_seen_remote_ip: destination_remote_ip,
+          account_id: account_id
+        },
+        id,
+        %Auth.Subject{
+          account: %{id: account_id},
+          actor: %{id: actor_id},
+          identity: %{id: identity_id},
+          expires_at: expires_at,
+          context: %Auth.Context{
+            remote_ip: source_remote_ip,
+            user_agent: source_user_agent
+          }
+        } = subject,
+        opts \\ []
+      ) do
+    with :ok <- Auth.ensure_has_permissions(subject, Authorizer.create_flows_permission()),
+         {:ok, resource} <- Resources.fetch_and_authorize_resource_by_id(id, subject, opts) do
+      flow =
+        Flow.Changeset.create(%{
+          policy_id: resource.authorized_by_policy.id,
+          client_id: client_id,
+          gateway_id: gateway_id,
+          resource_id: resource.id,
+          account_id: account_id,
+          source_remote_ip: source_remote_ip,
+          source_user_agent: source_user_agent,
+          destination_remote_ip: destination_remote_ip,
```

Comment:
Shouldn't this be the final resource IP as mapped by the Gateway? That may be tricky to implement (seems like the bulk of the flow logging feature will need to be reported from `connlib`) so maybe for now we just call this `gateway_remote_ip`?

Reply:
@jamilbk pushed a fix, also renamed `source_*` to `client_*` for consistency. 

Yes, if we want the rest of the metrics (like real traffic destinations) metrics are needed.

Reply:
👍